ubcpi/answer_pool.py

```diff
@@ -0,0 +1,315 @@
+import random
+import persistence as sas_api
+
+POOL_SIZE = 50
+
+
+class UnknownChooseAnswerAlgorithm(Exception):
+    pass
+
+
+def offer_answer(pool, answer, rationale, student_id, algo):
+    """
+    submit a student answer to the answer pool
+
+    The answer maybe selected to stay in the pool depending on the selection algorithm
+
+    Args:
+        pool (dict): answer pool
+            Answer pool format:
+            {
+                option1_index: {
+                    'student_id': { can store algorithm specific info here },
+                    ...
+                }
+                option2_index: ...
+            }
```

Comment:
@sarina it contains the integer option index (0,1 or 2) for the key of the dict and student Id for the key of sub dict. For current two algorithms, we don't store any algorithm specific info. So the value of the student id dict will be empty. But in the future, an algorithm may store a weight or score, which can be used for selection process. So that we don't need to store the the whole rationale there.